Miasma improvements

Elijahrane opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
Cleanup, bugfixes, lots of audiovisual detail.
**Screenshots**
<!-- If applicable, add screenshots to showcase your PR. If your PR is a visual change, add
screenshots or it's liable to be closed by maintainers. -->

**Changelog**
<!--
Here you can fill out a changelog that will automatically be added to the game when your PR is merged
There are 4 icons for changelog entries: add, remove, tweak, fix. I trust you can figure out the rest.

You can put your name after the :cl: symbol to change the name that shows in the changelog (otherwise it takes your GitHub username)
Like so: :cl: PJB

Generally, only put things in changelogs that players actually care about. Stuff like "Refactored X system, no changes should be visible" shouldn't be on a changelog.

For writing actual entries, don't consider the entry type suffix (e.g. add) to be "part" of the sentence:
bad: - add: a new tool for engineers
good: - add: added a new tool for engineers
-->

:cl: Rane
- add: Corpses are now circled by buzzing flies when they've started to rot.
- tweak: Do you know how heavy cows are? It's wack. They're expensive to transport, and the cow crate price has been increased.
- fix: Freezer crates now stop rotting.
- fix: Rejuvenation properly removes rotting.
- fix: Body parts should gib from rotting for real this time.

mirrorcult commented on Content.Server/Administration/Commands/RejuvenateCommand.cs:
```diff
@@ -98,6 +99,11 @@ public static void PerformRejuvenate(EntityUid target)
                 EntitySystem.Get<DiseaseSystem>().CureAllDiseases(target, carrier);
             }
+
+            if (entMan.TryGetComponent<RottingComponent>(target, out var rotting))
+            {
+                entMan.RemoveComponent<RottingComponent>(target);
+            }
```
```suggestion
            entMan.RemoveComponent<RottingComponent>(target);
```

remcomp checks for this already

mirrorcult commented on Resources/Prototypes/Entities/Mobs/NPCs/simplemob.yml:
```diff
@@ -151,6 +151,12 @@
   - type: MobPrice
     price: 1000 # Living critters are valuable in space.
   - type: Perishable
+  - type: AmbientSound #Ideally in the future this will support multiple sources
+    enabled: false
+    volume: -5
+    range: 3
+    sound:
+      path: /Audio/Ambience/Temporary/flies.ogg
```
yeah this is definitely sus right now. making the flies a separate entity is probably the best solution overall but need another opinion on that

Elijahrane replied:
Mmmm, would an entity that just holds the ambient sound stuff that gets parented to entities with FliesComponent and removed when it's shut down be decent?

mirrorcult commented on Content.Server/Atmos/Miasma/MiasmaSystem.cs:
```diff
@@ -77,36 +98,77 @@ private void OnGibbed(EntityUid uid, PerishableComponent component, BeingGibbedE
+        private void OnFliesInit(EntityUid uid, FliesComponent component, ComponentInit args)
+        {
+            component.VirtFlies = EntityManager.SpawnEntity("AmbientSoundSourceFlies", Transform(uid).Coordinates);
+            Transform(component.VirtFlies).ParentUid = uid;
```
```suggestion
            Transform(component.VirtFlies).AttachParent(uid);
```

mirrorcult commented on Resources/Audio/Ambience/Temporary/license.txt:
```diff
@@ -0,0 +1 @@
+flies.ogg taken from https://freesound.org/people/telezon/sounds/321870/
```
```suggestion
flies.ogg taken from https://freesound.org/people/telezon/sounds/321870/ licensed under CC-0
```

mirrorcult commented on Resources/Prototypes/Entities/Effects/ambient_sounds.yml:
```diff
@@ -0,0 +1,10 @@
+- type: entity
+  id: AmbientSoundSourceFlies
+  name: ambient sound source
+  suffix: flies
```
```suggestion
- type: entity
  id: AmbientSoundSourceFlies
  noSpawn: true
```

mirrorcult commented on Resources/Prototypes/Entities/Effects/ambient_sounds.yml:
```diff
@@ -0,0 +1,10 @@
+  name: ambient sound source
+  suffix: flies
+  components:
+  - type: AmbientSound
+    volume: -5
+    range: 3
+    sound:
+      path: /Audio/Ambience/Temporary/flies.ogg
```
```suggestion
  - type: AmbientSound
    volume: -5
    range: 3
    sound:
      path: /Audio/Ambience/Temporary/flies.ogg
  - type: Tag
    tags:
    - HideContextMenu
```

shouldn't happen normally but best to do this for all the abstract-type entities like this

mirrorcult commented on Resources/Prototypes/Entities/Mobs/Species/human.yml:
```diff
@@ -325,6 +325,12 @@
     price: 1500 # Kidnapping a living person and selling them for cred is a good move.
     deathPenalty: 0.01 # However they really ought to be living and intact, otherwise they're worth 100x less.
   - type: Perishable
+  - type: AmbientSound
+    enabled: false
+    volume: -5
+    range: 3
+    sound:
+      path: /Audio/Ambience/Temporary/flies.ogg
```
remove?

Elijahrane commented:
Addressed reviews + verified in game:
1. Sound plays when the entity starts to produce miasma and has fly visuals
2. Sound source is correctly parented to the entity
3. Sound source goes away after rejuvanate, entity deleted, or component removed
